### ISSUE TITLE
Dagger cannot be used as a Bazel repo dependency

### ISSUE THREAD

**Issue by Ubehebe** [google/dagger#1286]
- //java/dagger/internal/codegen:processor has a [dep](https://github.com/google/dagger/blob/master/java/dagger/internal/codegen/BUILD#L45) on @local_jdk//:lib/tools.jar.
- tools.jar [no longer exists](https://docs.oracle.com/javase/9/migrate/toc.htm#JSMIG-GUID-A78CC891-701D-4549-AA4E-B8DD90228B4B) in JDK 9+.
- Bazel 0.16 [ships](https://blog.bazel.build/2018/07/31/bazel-0.16.html) with an embedded JDK 9, and Bazel 0.17 [ships](https://blog.bazel.build/2018/09/14/bazel-0.17.html) with an embedded JDK 10.

Thus, it is no longer possible to build Dagger with Bazel 0.16+ with the default Java toolchain. (The error message is `missing input file '@local_jdk//:lib/tools.jar'`.)

**Comment by Ubehebe** [comment]
The scope of this issue is larger than I realized. I've changed the title to reflect that.

First, my description of how Bazel versions interact with JDK versions above was not quite right. There are _three_ JDKs used by Bazel, as @cushon describes in [this comment](https://github.com/bazelbuild/bazel/issues/5594#issuecomment-407169981). The JDK most relevant for Dagger is `--host_javabase`, which is used to run build-time tools (like Dagger's annotation processors). It defaults to Bazel's embedded JDK. That means that any tool dep on `@local_jdk` (like the one Dagger has) is inappropriate. It also means that Dagger is incompatible with using JDK9+ as the `@local_jdk` in _any_ version of Bazel, not just 0.16+.

I sent #1288 to replace the tools.jar dep. I think it's trivial. But I'm running into other, unrelated problems when trying to use Dagger HEAD as an upstream Bazel repo. I'll continue to link them to this issue as they arise.

I think Dagger could use better testing to verify that it can be consumed from Bazel (for example, cd83b30 broke this use case). Either that, or delete the Bazel integration and let downstream users curate their own BUILD rules.
